readthedocs/filetreediff/__init__.py

```diff
@@ -0,0 +1,84 @@
+import json
+from dataclasses import dataclass
+
+from readthedocs.builds.constants import BUILD_STATE_FINISHED
+from readthedocs.builds.models import Version
+from readthedocs.projects.constants import MEDIA_TYPE_METADATA
+from readthedocs.storage import build_media_storage
+
+
+@dataclass
+class FileTreeDiff:
+    added: list[str]
+    removed: list[str]
+    modified: list[str]
+
+
+def get_diff(version_a: Version, version_b: Version) -> FileTreeDiff | None:
```

Comment:
Would be good to get a docstring here talking more about the architecture. Basically we're saving the manifest during build time, and then querying the manifest when we're hitting the API endpoint to get the possible comparisons?